crates/rpc-types-engine/src/attributes.rs

```diff
@@ -70,11 +70,16 @@ impl OpPayloadAttributes {
     pub fn recovered_transactions_with_encoded(
         &self,
     ) -> impl Iterator<Item = WithEncoded<Recovered<OpTxEnvelope>>> + '_ {
-        self.transactions
-            .iter()
-            .flatten()
-            .zip(self.recovered_transactions())
-            .map(|(tx_bytes, recovered)| WithEncoded::new(tx_bytes.clone(), recovered))
+        self.transactions.iter().flatten().filter_map(|tx_bytes| {
+            let env = OpTxEnvelope::decode_2718(&mut tx_bytes.as_ref()).ok()?;
+            let txenv = env.try_into_eth_envelope().ok()?;
+            let recovered = txenv.try_into_recovered().ok()?;
```

Comment:
this code can be moved into a separate function `pub fn try_into_recovered(bytes: &[u8]) -> Result<RecoveredType, ErrorType>` and called here and in the body of the method below `recovered_transactions`